music/src/core/melodies.ts

```diff
@@ -144,3 +145,112 @@ export class Melody {
     return noteSequence;
   }
 }
+
+/**
+ * MelodyControl interface for extracting control signals from melodies. These
+ * control signals can be used as conditioning for melody generation, or can
+ * themselves be generated.
+ *
+ * @param depth The dimension of the control signal at each step.
+ */
+export interface MelodyControl {
+  readonly depth: number;
+  extract(melody: Melody): tf.Tensor2D;
+}
+
+/**
+ * Rhythm control signal. Extracts a depth-1 tensor indicating note onsets.
+ */
+export class MelodyRhythm implements MelodyControl {
+  readonly depth = 1;
```

Comment:
i think this can just be private then? does anyone need to actually read it?

Reply:
It's the dimension of the returned arrays, which really feels to me like something that should be public.